File: src/main/com/intellij/lang/jsgraphql/schema/library/GraphQLLibraryDescriptor.java
package com.intellij.lang.jsgraphql.schema.library;

import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class GraphQLLibraryDescriptor {

  private final String myIdentifier;

  public GraphQLLibraryDescriptor(@NotNull String identifier) {
    myIdentifier = identifier;
  }

  public @NotNull String getIdentifier() {
    return myIdentifier;
  }

  public @NotNull String getPresentableText() {
    return getIdentifier();
  }

  public boolean isEnabled(@NotNull Project project) {
    return true;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    GraphQLLibraryDescriptor that = (GraphQLLibraryDescriptor)o;
    return Objects.equals(myIdentifier, that.myIdentifier);
  }

  @Override
  public int hashCode() {
    return Objects.hash(myIdentifier);
  }

  @Override
  public String toString() {
    return "GraphQLLibraryDescriptor{" +
           "myIdentifier='" + myIdentifier + '\'' +
           '}';
  }
}

File: src/main/com/intellij/lang/jsgraphql/schema/library/GraphQLLibraryDescriptor.kt
package com.intellij.lang.jsgraphql.schema.library

import com.intellij.openapi.project.Project

abstract class GraphQLLibraryDescriptor(
  val identifier: String,
  val displayName: String = identifier,
  val description: String? = null,
  val attachmentScope: GraphQLLibraryAttachmentScope = GraphQLLibraryAttachmentScope.GLOBAL,
) {
  abstract fun isEnabled(project: Project): Boolean

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (javaClass != other?.javaClass) return false

    other as GraphQLLibraryDescriptor

    if (identifier != other.identifier) return false
    if (displayName != other.displayName) return false
    if (description != other.description) return false
    if (attachmentScope != other.attachmentScope) return false

    return true
  }

  override fun hashCode(): Int {
    var result = identifier.hashCode()
    result = 31 * result + displayName.hashCode()
    result = 31 * result + (description?.hashCode() ?: 0)
    result = 31 * result + attachmentScope.hashCode()
    return result
  }
}
